packages/integration-sdk-runtime/src/synchronization/index.ts

```diff
@@ -319,14 +333,10 @@ function handleUploadDataChunkError({
    */
   const systemErrorResponseData = getSystemErrorResponseData(err);
 
-  if (err.code === 'RequestEntityTooLargeException') {
-    // No reason to retry these errors as the request size ain't gonna change.
-    throw new IntegrationError({
-      code: 'INTEGRATION_UPLOAD_FAILED',
-      cause: err,
-      fatal: false,
-      message: 'Failed to upload integration data because payload is too large',
-    });
+  if (isRequestUploadTooLargeError(err)) {
```

Comment:
If the upload batch is a batch of relationships, we should bubble `isRequestUploadTooLargeError`.

Reply:
I added a check that if we don't have any Entities and/or we don't have any _rawData entries in those Entities that we can truncate it throws.